src/sync.h

```diff
@@ -85,33 +87,35 @@ void static inline DeleteLock(void* cs) {}
 #define AssertLockHeld(cs) AssertLockHeldInternal(#cs, __FILE__, __LINE__, &cs)
 
 /**
- * Wrapped boost mutex: supports recursive locking, but no waiting
+ * Wrapped mutex: supports recursive locking, but no waiting
  * TODO: We should move away from using the recursive lock by default.
  */
-class CCriticalSection : public AnnotatedMixin<boost::recursive_mutex>
+class CCriticalSection : public AnnotatedMixin<std::recursive_mutex>
 {
 public:
     ~CCriticalSection() {
         DeleteLock((void*)this);
     }
 };
 
-/** Wrapped boost mutex: supports waiting but not recursive locking */
-typedef AnnotatedMixin<boost::mutex> CWaitableCriticalSection;
+/** Wrapped mutex: supports waiting but not recursive locking */
+typedef AnnotatedMixin<std::mutex> CWaitableCriticalSection;
 
-/** Just a typedef for boost::condition_variable, can be wrapped later if desired */
-typedef boost::condition_variable CConditionVariable;
+/** Just a typedef for std::condition_variable, can be wrapped later if desired */
+typedef std::condition_variable CConditionVariable;
+
+/** Just a typedef for std::unique_lock, can be wrapped later if desired */
+typedef std::unique_lock<std::mutex> Lock;
```

Comment:
I'd really rather we *not* call this a "Lock" - its not intended to be the thing used across the codebase as any lock (though we could maybe do so if we add DEBUG_LOCKORDER support here).

Reply:
You always say what you don't want instead of saying what you do want ;). What would you like instead? If you've seen https://github.com/bitcoin/bitcoin/pull/11599, the name should make more sense in this context. I think Lock and Mutex are the right names because they match the standard. As you mentioned, you can implement DEBUG_LOCKORDER with them and also do other things useful for debugging, like printing differences in a data structure between time a particular mutex is locked & unlocked.

Reply:
The original name (CWaitableCriticalSection) was nice and specific - I'm happy to make it more in-line with the more-accepted "lock" terminology, but pointing out that it has only a narrow use-case in our codebase is important.

Reply:
Ok, but I don't think non-recursive locks should have a narrow usecase in our codebase. I thought everybody agreed non-recursivelocks should be preferred to recursive locks.

Reply:
Also CWaitableCriticalSection was not the original name for this. There was no original name for this. Previously code just used boost::lock directly.

Reply:
Heh, I meant in the original patch in this PR. I mean we use locks in lots of different ways, for new subsystems that have very careful locking models, obviously non-recursive locks are preferred, but I'm skeptical every new thing we add will use them.

Reply:
@ryanofsky @TheBlueMatt Could you provide a diff with your suggested change?

Reply:
I think Matt basically wants the typedef to be called ScaryLock instead of Lock. But I'm fine with whatever is proposed here given there is already a PR open for renaming.

Reply:
OK, I'll fix. @TheBlueMatt, what is your suggested name instead of `Lock`? :-)

Reply:
I was fine with the WaitableCriticalSection (or WaitableLock or whatever) version that was there originally.

Reply:
@TheBlueMatt @ryanofsky The following diff would work?

```diff
diff --git a/src/init.cpp b/src/init.cpp
index dd64602..9e3eb6b 100644
--- a/src/init.cpp
+++ b/src/init.cpp
@@ -551,7 +551,7 @@ static void BlockNotifyGenesisWait(bool, const CBlockIndex *pBlockIndex)
 {
     if (pBlockIndex != nullptr) {
         {
-            Lock lock_GenesisWait(cs_GenesisWait);
+            WaitableLock lock_GenesisWait(cs_GenesisWait);
             fHaveGenesis = true;
         }
         condvar_GenesisWait.notify_all();
@@ -1634,7 +1634,7 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)

     // Wait for genesis block to be processed
     {
-        Lock lock(cs_GenesisWait);
+        WaitableLock lock(cs_GenesisWait);
         while (!fHaveGenesis) {
             condvar_GenesisWait.wait(lock);
         }
diff --git a/src/rpc/mining.cpp b/src/rpc/mining.cpp
index 8fae01b..0ba0e96 100644
--- a/src/rpc/mining.cpp
+++ b/src/rpc/mining.cpp
@@ -478,7 +478,7 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
         {
             checktxtime = std::chrono::steady_clock::now() + std::chrono::minutes(1);

-            Lock lock(csBestBlock);
+            WaitableLock lock(csBestBlock);
             while (chainActive.Tip()->GetBlockHash() == hashWatchedChain && IsRPCRunning())
             {
                 if (cvBlockChange.wait_until(lock, checktxtime) == std::cv_status::timeout)
diff --git a/src/sync.h b/src/sync.h
index 66b6d96..20556af 100644
--- a/src/sync.h
+++ b/src/sync.h
@@ -105,7 +105,7 @@ typedef AnnotatedMixin<std::mutex> CWaitableCriticalSection;
 typedef std::condition_variable CConditionVariable;

 /** Just a typedef for std::unique_lock, can be wrapped later if desired */
-typedef std::unique_lock<std::mutex> Lock;
+typedef std::unique_lock<std::mutex> WaitableLock;

 #ifdef DEBUG_LOCKCONTENTION
 void PrintLockContention(const char* pszName, const char* pszFile, int nLine);
```

Reply:
> @TheBlueMatt @ryanofsky The following diff would work?

Yes for me. (As I mentioned in my last comment I am happy with any naming in this pr since #11599 can clean up any inconsistencies.)

Reply:
@ryanofsky Excellent! OK with you @TheBlueMatt? :-)

Reply:
Yea, looks good, thanks!